packages/contentful--app-scripts/lib/clean-up/clean-up-bundles.js

```diff
@@ -29,20 +32,62 @@ const scheduleBundleDeletion = async (bundlesToDelete, client, settings) => {
 };
 
 async function cleanUpBundles(settings) {
-  let bundles, definition;
+  let bundlesToDelete, definition;
+
   const client = createClient(
     {
       host: settings.host,
       accessToken: settings.accessToken,
     },
     { type: 'plain' }
   );
-  const bundlesSpinner = ora(`Fetching all bundles...`).start();
-  try {
-    bundles = await client.appBundle.getMany({
+
+  const fetchAppBundles = async (limit = DEFAULT_BUNDLES_TO_FETCH, skip = 0) => {
+    return await client.appBundle.getMany({
       appDefinitionId: settings.definition.value,
       organizationId: settings.organization.value,
+      query: {
+        limit,
+        skip,
+      },
     });
+  };
+
+  const getAppBundles = async (requestedAmount = DEFAULT_BUNDLES_TO_FETCH) => {
+    if (requestedAmount < 1) {
+      throw new Error('Requested amount of bundles to fetch must be greater than 0');
+    }
+    const getBundles = async (limit, skip) => {
+      const result = await fetchAppBundles(limit);
+      const currLength = skip + result.items.length;
+
+      if (currLength <= requestedAmount && result.total > currLength) {
+        return [...result.items, ...(await getBundles(result.items.length, currLength))];
+      } else {
+        return result.items;
+      }
+    };
+
+    const all = await getBundles(DEFAULT_BUNDLES_TO_FETCH, 0);
+
+    return all.slice(0, requestedAmount);
+  };
+
+  const getAppBundleCount = async () => {
+    return await fetchAppBundles(1, 0);
+  };
+
+  const bundlesSpinner = ora(`Fetching all bundles...`).start();
+  try {
+    const { total } = await getAppBundleCount();
+    const amountToDelete = total - settings.keep;
+    if (amountToDelete < 1) {
+      console.log(`${chalk.yellow('Warning:')} There is nothing to delete`);
+      bundlesSpinner.stop();
+      return;
+    }
+    bundlesToDelete = await getAppBundles(amountToDelete);
```

Comment:
As a quick fix, we could just sort them by their `sys.createdAt` information - wdyt?

Reply:
I tried that but it seems the sorting didn't work. I have a call to check this scheduled today

Reply:
Ah, what I meant is, sorting your local list, since you're fetching all bundles, you can sort them locally, and just delete the oldest bundles.